### DISCUSSION TITLE
Dataloader: field or type missing in schema

### DISCUSSION THREAD

**Discussion by tknupf** [Netflix/dgs-framework#1898] [Q&A]
I have implemented a first DataLoader.
I also created the corosponing method in the DataFetcher:

```
   @DgsData(parentType = "A")
    public CompletableFuture<C> b(DgsDataFetchingEnvironment dfe) {
        DataLoader<UUID, B> dataLoader = dfe.getDataLoader(B.class);
        return dataLoader.load(dfe.getArgument("id"), dfe.getLocalContext());
    }
```

But when I start the service:
`com.netflix.graphql.dgs.exceptions.DataFetcherSchemaMismatchException: @DgsData in com.daimler.ltm.api.container.datafetchers.ADataFetcher on field images references object type `A` it has no field named `b`. All data fetchers registered with @DgsData|@DgsQuery|@DgsMutation|@DgsSubscription must match a field in the schema.`

When I insert the field in the type A then I am missing the type B:
```
type A {
  ...
  b: [B]
}
```

What am I doing wrong?

**Comment by paulbakker** [comment]
Your example looks fine - Double check that the schema file is actually picked up correctly, and it's not getting it from a place different than what you're looking at.

**Reply by tknupf** [reply]
I managed to fix it by inserting in my schema:
```
type A {
  ...
  b: [B]
}

type B @key(fields: "id") @external {
   id: ID!
   ....
}
```
Do I really have to duplicate the Type B witch is from a different service?
